query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -596,6 +606,9 @@ private void handleTableFormatResultSet(
             colIdx++;
           } else {
             String val = resultSet.getString(rowIdx, colIdx);
+            if(selectionMap.containsKey(colIdx) && selectionMap.get(colIdx).equals("AVG_RATE")){
```

Comment:
Just check if the colIdx exists in avgRate set return by `getAvgRateColumnIndices` and apply reverse transformation.